client/js/components/StatsLeaderboard.vue

```diff
@@ -0,0 +1,359 @@
+<template>
+  <div>
+    <div id="stats-table-upper-surface" class="d-flex align-end">
+      <v-select
+        v-model="selectedMetric"
+        :items="metricChoices"
+        label="Select Metric"
+        class="filter-select mr-4 fit"
+        data-cy="metric-select"
+      />
+      <v-select
+        v-model="selectedWeeks"
+        :items="weeks"
+        label="Select Weeks"
+        data-cy="week-select"
+        class="flex-grow-1"
+        menu-props="data-week-select-menu"
+        multiple
+        chips
+        deletable-chips
+      />
+    </div>
+    <v-data-table
+      :items="tableRows"
+      :headers="tableColumns"
+      :sort-by.sync="sortBy"
+      :loading="loading"
+      :item-class="tableRowClass"
+    >
+      <template #[`item.rank`]="{item, value}">
+        <span :data-rank="item.username">{{ value }} </span>
+      </template>
+      <template v-for="week in ['total', ...selectedWeeks]" #[`item.week_${week}`]="{item, value}">
```

Comment:
Bit of fancy footwork here. This customizes each column with the name `week_total`, or `week_n`. `item` refers to the entire row being rendered and `value` is the value associated with the specific cell being customized, which will be the number of points the player received for this week (that's what's used for sorting purposes)

Reply:
This could probably use a comment